lifelines/plotting.py

```diff
@@ -433,6 +434,113 @@ def add_at_risk_counts(*fitters, **kwargs):
     return ax
 
 
+def plot_interval_censored_lifetimes(
+    lower_bound,
+    upper_bound,
+    entry=None,
+    left_truncated=False,
+    sort_by_lower_bound=True,
+    event_observed_color="#A60628",
+    event_right_censored_color="#348ABD",
+    ax=None,
+    **kwargs
+):
+    """
+    Returns a lifetime plot for interval censored data.
+
+    Parameters
+    -----------
+    lower_bound: (n,) numpy array or pd.Series
+      the start of the period the subject experienced the event in.
+    upper_bound: (n,) numpy array or pd.Series
+      the end of the period the subject experienced the event in. If the value is equal to the corresponding value in lower_bound, then
+      the individual's event was observed (not censored).
+    entry: (n,) numpy array or pd.Series
+      offsetting the births away from t=0. This could be from left-truncation, or delayed entry into study.
+    left_truncated: boolean
+      if entry is provided, and the data is left-truncated, this will display additional information in the plot to reflect this.
+    sort_by_lower_bound: boolean
+      sort by the lower_bound vector
+    event_observed_color: str
+      default: "#A60628"
+    event_right_censored_color: str
+      default: "#348ABD"
+      applies to any individual with an upper bound of infinity.
+
+    Returns
+    -------
+    ax:
+
+    Examples
+    ---------
+    .. code:: python
+
+        import pandas as pd
+        import numpy as np
+        from lifelines.plotting import plot_interval_censored_lifetimes
+        df = pd.DataFrame({'lb':[20,15,30, 10, 20, 30], 'ub':[25, 15, np.infty, 20, 20, np.infty]})
+        ax = plot_interval_censored_lifetimes(lower_bound=df['lb'], upper_bound=df['ub'])
+    """
+
+    def iloc(x, i):
```

Comment:
I think we should pull this out into a top level function so both `plot_..._lifetimes` function can use it.

Reply:
Yes, good point, I've moved it.